charmtools/build/builder.py

```diff
@@ -303,28 +303,32 @@ def fetch_dep(self, layer, results):
                 results["layers"].append(base_layer)
                 self.post_metrics('layer', base_layer.name, base_layer.fetched)
 
-    def build_tactics(self, entry, layer, next_config, output_files):
+    def build_tactics(self, entry, layer, next_config, current_config,
+                      output_files):
         relname = entry.relpath(layer.directory)
         existing_tactic = output_files.get(relname)
         tactic = Tactic.get(
             entry,
             self.target,
             layer,
             next_config,
+            current_config,
             existing_tactic,
         )
         output_files[relname] = tactic
 
     def plan_layers(self, layers, output_files):
+        current_config = BuildConfig()
         next_config = BuildConfig()
```

Comment:
Instead of calling BuildConfig() twice, would it be more efficient to simply say:
next_config = current_config